randomwallpaper@iflow.space/utils.js

```diff
@@ -0,0 +1,74 @@
+const Gdk = imports.gi.Gdk;
+const Gio = imports.gi.Gio;
+const GLib = imports.gi.GLib;
+
+var Utils = class {
+	static getRandomNumber(size) {
+		return Math.floor(Math.random() * size);
+	}
+
+	// https://gjs.guide/guides/gio/subprocesses.html#waiting-for-processes
+	static runCommand(argv) {
+		return new Promise((resolve, reject) => {
+			try {
+				let proc = Gio.Subprocess.new(argv, Gio.SubprocessFlags.NONE);
+
+				proc.wait_async(null, (proc, result) => {
+					try {
+						resolve(proc.wait_finish(result));
+					} catch (error) {
+						reject(error);
+					}
+				});
+			} catch (error) {
+				reject(error);
+			}
+		});
+	}
+
+	// https://gjs.guide/guides/gio/subprocesses.html#complete-examples
+	/**
+	 * Execute a command asynchronously and check the exit status.
+	 *
+	 * If given, @cancellable can be used to stop the process before it finishes.
+	 *
+	 * @param {string[]} argv - a list of string arguments
+	 * @param {Gio.Cancellable} [cancellable] - optional cancellable object
+	 * @returns {Promise<>} - The process success
+	 */
+	static async #execCheck(argv, cancellable = null) {
```

Comment:
Both functions here for executing stuff commands aren't used in this commit.
1. Will they be used later on?
2. The hashtag as a prefix for private class members is not supported in GJS, or is it?

Reply:
> 1\. Will they be used later on?

I figured them too equal and ditched `runCommand` later on. Instead I'll call `execCheck` every time and don't care about the return value if necessary. `execCheck` will be public and will be used for all external program calls.

> 2\. The hashtag as a prefix for private class members is not supported in GJS, or is it?

Since ES6 modules are supported I thought they should be functional. At least the IDE will notice.
This was bad habit anyway and this will change to the recommended underscore variant.

Reply:
Alright, thanks! :)